src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -2452,12 +2437,106 @@ public static unsafe bool TryToBase64Chars(ReadOnlySpan<byte> bytes, Span<char>
+            Debug.Assert(chars.Length >= charLengthRequired);
+
+            // Base64-encode the bytes directly into the destination char buffer (reinterpreted as a byte buffer).
+            OperationStatus status = Base64.EncodeToUtf8(bytes, MemoryMarshal.AsBytes(chars), out _, out int bytesWritten);
+            Debug.Assert(status == OperationStatus.Done && charLengthRequired == bytesWritten);
+
+            // Now widen the ASCII bytes in-place to chars (if the vectorized ASCIIUtility.WidenAsciiToUtf16 is ever updated
+            // to support in-place updates, it should be used here instead). Since the base64 bytes are all valid ASCII, the byte
+            // data is guaranteed to be 1/2 as long as the char data, and we can widen in-place.
+            ref char dest = ref MemoryMarshal.GetReference(chars);
+            ref byte src = ref Unsafe.As<char, byte>(ref dest);
+            ref byte srcBeginning = ref src;
+
+            // We process the bytes/chars from right to left to avoid overwriting the remaining unprocessed data.
+            // The refs start out pointing just past the end of the data, and each iteration of a loop bumps
+            // the refs back the apropriate amount and performs the copy/widening.
+            dest = ref Unsafe.Add(ref dest, charLengthRequired);
+            src = ref Unsafe.Add(ref src, charLengthRequired);
+
+            // Handle 32 bytes at a time.
+            if (Vector256.IsHardwareAccelerated)
+            {
+                ref byte srcBeginningPlus31 = ref Unsafe.Add(ref srcBeginning, 31);
+                while (Unsafe.IsAddressGreaterThan(ref src, ref srcBeginningPlus31))
+                {
+                    src = ref Unsafe.Subtract(ref src, 32);
+                    dest = ref Unsafe.Subtract(ref dest, 32);
+
+                    Vector256<byte> utf8Vector = Vector256.LoadUnsafe(ref src);
+                    (Vector256<ushort> utf16Lower, Vector256<ushort> utf16Upper) = Vector256.Widen(utf8Vector);
+
+                    utf16Lower.StoreUnsafe(ref Unsafe.As<char, ushort>(ref dest));
+                    utf16Upper.StoreUnsafe(ref Unsafe.As<char, ushort>(ref dest), 16);
+                }
+            }
+
+            // Handle 16 bytes at a time.
+            if (Vector128.IsHardwareAccelerated)
+            {
+                ref byte srcBeginningPlus15 = ref Unsafe.Add(ref srcBeginning, 15);
+                while (Unsafe.IsAddressGreaterThan(ref src, ref srcBeginningPlus15))
+                {
+                    src = ref Unsafe.Subtract(ref src, 16);
+                    dest = ref Unsafe.Subtract(ref dest, 16);
+
+                    Vector128<byte> utf8Vector = Vector128.LoadUnsafe(ref src);
+                    (Vector128<ushort> utf16Lower, Vector128<ushort> utf16Upper) = Vector128.Widen(utf8Vector);
+
+                    utf16Lower.StoreUnsafe(ref Unsafe.As<char, ushort>(ref dest));
+                    utf16Upper.StoreUnsafe(ref Unsafe.As<char, ushort>(ref dest), 8);
```

Comment:
Nit: you can remove 4 `Unsafe.As<char` if you make `dest` `ref ushort` in its definition